### DISCUSSION TITLE
PocketBase v23.11 How to retrieve the authenticated user.

### DISCUSSION THREAD

**Discussion by razsa** [pocketbase/pocketbase#6157] [Q&A]
```
curl -X POST http://localhost:8090/api/orders -H "Content-Type: application/json" -H "Authorization: YOUR_AUTH_TOKEN" -d '{
>     "cartItems": [
>         {
>             "productId": "PRODUCT_ID_HERE",
>             "quantity": 2
>         },
>         {
>             "productId": "ANOTHER_PRODUCT_ID",
>             "quantity": 1
>         }
>     ],
>     "shippingAddress": "123 Test Street, Test City, 12345"
> }'
{"error":"Authentication required"}
```
```
// Create order endpoint
se.Router.POST("/api/orders", func(e *core.RequestEvent) error {
	// Require authentication
	authRecord, _ := e.Get("authRecord").(*core.Record)
	if authRecord == nil {
	       return e.JSON(http.StatusUnauthorized, map[string]string{"error": "Authentication required"})
	}
```

**Comment by ganigeorgiev** [comment]
_**If you used LLM to generate the above code sample, please stop doing that!**_

There is no such thing. Please read the documentation, and more specifically https://pocketbase.io/docs/go-routing/.

If you are trying to upgrade to v0.23+ from an older version, then please read https://pocketbase.io/v023upgrade/go/ (see also the [CHANGELOG](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#v0230) for overview).

In any case, if you need further help, try to elaborate more on your question and problem because the above title and description doesn't really make much sense to me.

**Reply by razsa** [reply]
I'm upgrading from an older version of PocketBase. But I never used the Go as framework mode. I'm trying to understand it.

I made 3 custom endpoints.
The endpoints requires auth but I don't understand how get the auth token to actually POST at 
the endpoint `/api/orders`.

**Reply by ganigeorgiev** [reply]
If your 3 custom endpoints look like in the above example, then they are incorrect. Again, please spend some time reading the documentation and upgrade guide.

The auth state can be accessed via `e.Auth` (see https://pocketbase.io/docs/go-routing/#retrieving-the-current-auth-state)
